airflow/serialization/serialized_objects.py

```diff
@@ -707,6 +710,9 @@ def deserialize_operator(cls, encoded_op: Dict[str, Any]) -> Union[BaseOperator,
                 v = cls._deserialize_timedelta(v)
             elif k in encoded_op["template_fields"]:
                 pass
+            elif k == "resources" and Resources.is_operator_resources_json_string(v):
```

Comment:
Can you explain why this` is_...()` function  is needed? It's not a pattern we have elsewhere in serialization.

Reply:
`# there are other values with the same key as `resources` so adding this check`, with more thoughts, i think we can remove it.